### PR TITLE
Add `target PodTemplate` which exposes the full Pod (not only the spec)

### LINKED ISSUE #735: Add `target: PodTemplate`
**Is your feature request related to a problem? Please describe.**
I can write a schema that targets `PodSpec`, but that doesn't allow me to match on the metadata in the pod template

**Describe the solution you'd like**
Add a new `PodTemplate` target similar to `PodSpec`

**Describe alternatives you've considered**
Write a really complicated schema with `target: Controller` that drills into the PodTemplate (different path for Cronjob vs deployment etc)

**Additional context**
Requested in Community Slack

### PR DESCRIPTION
**Pull request by ivanfetch-fw**
This PR fixes #735 

## Checklist
* [x] I have signed the CLA
* [ ] I have updated/added any relevant documentation

### What's the goal of this PR?
Add a `target: PodTemplate` to make the full pod (not only the spec) available to schema-checking.

### What changes did you make?
Added a new target `PodTemplate`, a function `GetPodTemplate` to return the complete pod containing `spec.containers`, and updated other logic to process the new `target: PodTemplate`.

### CODE REVIEW

**Review by makoscafee — APPROVED** [review]
LGTM. i just did not understand why the `SchemaTestCase.Target` has to be TargetPodSpec when you want `PodTemplateSpec`

File: pkg/config/schema.go [hunk 1]
```diff
@@ -304,7 +312,12 @@ func (check SchemaCheck) CheckAdditionalObjects(groupkind string, objects []inte
 // IsActionable decides if this check applies to a particular target
 func (check SchemaCheck) IsActionable(target TargetKind, kind string, isInit bool) bool {
 	if funk.Contains(HandledTargets, target) {
-		if check.Target != target {
+		if check.Target != TargetPodTemplate && check.Target != target {
```
[the remark this answers is not in the document]
Comment (ivanfetch-fw) [reply pkg/config/schema.go:315]: @makoscafee  RE: your suggestion, if we keep `if check.Target != target {` this function will return false; short-circuiting processing for PodTemplate.

The reason I say this is that the `applyPodSchemaChecks` function in pkg/validator/schema.go explicitly sets `schemaTestCase.Target` to `TargetPodSpec`. This causes the Target in `schemaTestCase` to NOT match the Target passed into this `IsActionable(` function.

I'm also happy to get on a Zoom to chat about this.

[the remark this answers is not in the document]
Comment (makoscafee) [reply pkg/config/schema.go:315]: makes sense what if 
`if check.Target != target {` is after processing for podTemplate?

[the remark this answers is not in the document]
Comment (makoscafee) [reply pkg/config/schema.go]: that sounds good.

**Pull request merged**